common/hexutil/hexutil.go

```diff
@@ -64,10 +64,6 @@ func Decode(input string) ([]byte, error) {
 	if !has0xPrefix(input) {
 		return nil, ErrMissingPrefix
 	}
-	// better compatible with odd size string
```

Comment:
@noel2004  @HAOYUatHZ  Why add these codes before? It causes some tests failed.

Reply:
I guess it require the handled hex strings has '0x' prefix for some legacy reasons? Anyway, enforcing the prefix should has some benefits in developing for avoiding decoding dec or some other type of strings by mistake. So I advice to keep it unless there are too many difficulities for making a workaround.

Reply:
Keep these codes will cause error.  Here it is https://github.com/scroll-tech/go-ethereum/actions/runs/3794039841/jobs/6451737909  
Failed: 
   - signed_data_internal_test.go:105:   TestParseBytes
  
If keep them, must fix the `TestParseBytes`.